com.unity.mobile.notifications/Runtime/iOS/iOSNotification.cs

```diff
@@ -380,7 +390,24 @@ internal iOSNotificationWithUserInfo GetDataForSending()
             iOSNotificationWithUserInfo ret;
             ret.data = data;
             ret.userInfo = userInfo;
+            ret.attachments = Attachments;
             return ret;
         }
+
+        /// <summary>
+        /// Add an attachment to notification.
+        /// </summary>
+        /// <param name="attachment">Attachment to add</param>
+        public void AddAttachment(iOSNotificationAttachment attachment)
+        {
+            var attachments = Attachments;
```

Comment:
I don't understand why we need AddAttachment  function?

Maybe initialize Attachments in in iOSNotification constructor, then as I user I can simply do

var notification = new iOSNotification ();
notifacation.Attachements.Add(...)

Reply:
I don't expect attachments to be frequently used, so I'd like to avoid the allocation. So I added this for convenience.

Reply:
Even so, this looks wrong, since you're providing with two ways on how to add attachements, if you're worried about allocations, why not do this then:
* Remove AddAttachment
* Change public List<iOSNotificationAttachment> Attachments { get; set; } to
```
List<iOSNotificationAttachment> m_Attachements;
public List<iOSNotificationAttachment> Attachments {
{
get 
{
   if (m_Attachements == null)
       m_Attachements = new List<iOSNotificationAttachment>();
   return m_Attachements ; 
}
set
{
   m_Attachements  = value;
}

}
```

Reply:
But this make querying attachments create a list.
From my expected usage AddAttachment would be enough, but I don't want to limit users to not being able remove that added attachments.

Reply:
I am not sure I understand why you're trying so hard here to avoid allocation, since as far as I understand, attaching attachements is a very rare operation, on other hand, you're creating this inconsistent API, where you have AddAttachement function and yet you also have a property where you can also manipulate the list of attachements..

Here's another suggestion you could do:
* Add RemoveAttachement
* Change public List<iOSNotificationAttachment> Attachments to IReadOnlyList<iOSNotifcationAttachemen>

That way there wouldn't duplication of operations on API side.


But in reality, I would simply leave List<iOSNotificationAttachment> Attachments property and initialize it in iOSNotification,  since user needs to instantiate iOSNotification anyways, right? This would avoid this all unecessary complexity in your code